rust/integration-tests/src/event_processor/event_processor_tests.rs

```diff
@@ -0,0 +1,255 @@
+    #[tokio::test]
+    async fn test_multiple_account_transfers_with_correct_withdraw_events() {
+        let test_context = TestContext::new(&[
+            ACCOUNT_A_GET_FUNDED_FROM_FAUCET,
+            ACCOUNT_A_TRANSFER_TO_ACCOUNT_B,
+            ACCOUNT_B_TRANSFER_TO_ACCOUNT_B,
+        ])
+        .await
+        .unwrap();
+
+        assert!(test_context
+            .run(default_processor_config(), |conn: &mut PgConnection| {
+                let result = events
+                    .select((
+                        transaction_version,
+                        event_index,
+                        type_,
+                        data,
+                        creation_number,
+                    ))
+                    .filter(transaction_version.eq(0))
+                    .load::<(i64, i64, String, serde_json::Value, i64)>(conn);
+                let result_events = result.expect("Failed to load events");
+                assert_eq!(result_events.len(), 5);
+
+                let withdraw_events = events
+                    .select((transaction_version, event_index, type_))
+                    .filter(type_.eq("0x1::coin::WithdrawEvent"))
+                    .load::<(i64, i64, String)>(conn);
+
+                // Withdraw from Faucet, Withdraw from AccountA, Withdraw from AccountB
+                assert_eq!(withdraw_events.unwrap().len(), 3);
+
+                let deposit_events = events
+                    .select((
+                        transaction_version,
+                        event_index,
+                        type_,
+                        data,
+                        account_address,
+                    ))
+                    .filter(type_.eq("0x1::coin::DepositEvent"))
+                    .load::<(i64, i64, String, serde_json::Value, String)>(conn);
+
+                let deposit_events = deposit_events.expect("Failed to load DepositEvent");
+                let account_a_address =
+                    "0xeea01e0c163fe390e30afd0e6ff88a3535a2d78b4cbbcc8f9bf3848b5a8fbcf2";
+                // Verify that specific events with a given transaction version and creation number contain the correct data
+                deposit_events
+                    .iter()
+                    .filter(|(_, _, _, _, _account_address)| _account_address.eq(account_a_address)) // Ensure correct creation number
+                    .for_each(|(_, _, _, _data, _)| {
+                        // Ensure that any key starting with "amount" has the expected value
+                        _data
+                            .as_object()
+                            .unwrap()
+                            .iter()
+                            .filter(|(k, _)| k.starts_with("amount"))
+                            .for_each(|(_, v)| {
+                                assert_eq!(v.as_str().unwrap(), "100000000");
```

Comment:
does this mean all deposit amount is 100000000?

Reply:
for it's testing against specific event filtered by `account_a_address`. I am thinking of adding a feature of using input json to extract values so then we can easily compare all data

Reply:
probably you can add in comments like `checking deposit into account A is 100000000`